contracts/contribute/Contributors.sol

```diff
@@ -530,39 +538,44 @@ contract Contributors is ERC721TokenReceiver {
         IStaking(stakingInstance).unstake(serviceId);
 
         // Terminate service
-        (, uint256 terminateRefund) = IService(serviceManager).terminate(serviceId);
+        (, uint256 refund) = IService(serviceManager).terminate(serviceId);
+        uint256 refundNative = 1;
 
-        // Unbond service
-        (, uint256 unbondRefund) = IService(serviceManager).unbond(serviceId);
-
-        // Calculate refund
-        uint256 refund = terminateRefund + unbondRefund;
+        // Unbond service, if operator is address(this)
+        if (IService(serviceRegistry).mapAgentInstanceOperators(msg.sender) == address(this)) {
+            (, uint256 unbondAmount) = IService(serviceManager).unbond(serviceId);
+            refund += unbondAmount;
+            refundNative += NUM_AGENT_INSTANCES;
+        }
```

Comment:
After unstaking, terminating and unbonding the service, the user is able to deploy the service themselves with a different operator. In this scenario the CM is not responsible for the service unbonding when it gets unstaked again.